### PR TITLE
 categories cards container

### LINKED ISSUE #24: CategoriesCardsContainer Component
## Acceptance Criteria
 
 - [ ] Props
 - [ ] Components
 - [ ] Snapshot

-----------------------------------------------------------------------
### Props
- categories 
- loading
-----------------------------------------------------------------------
### Components
- Title Component
- CategoriesCard Component
- Spinner
- Empty

-----------------------------------------------------------------------
### Snapshot
image

### PR DESCRIPTION
**Pull request by IsraaHamdi**
RELATES #24 
Close #24
image

### CODE REVIEW

File: client/src/App/index.jsx [hunk 1]
```diff
@@ -8,20 +8,23 @@ import '../style/custom-antd.css';
 import './app.css';
 
 function App() {
+  const [loading, setLoaging] = useState(false);
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:11]:
```suggestion
  const [loading, setLoading] = useState(false);
```

File: client/src/App/index.jsx [hunk 2]
```diff
@@ -8,20 +8,23 @@ import '../style/custom-antd.css';
...
   const [categories, setCategories] = useState([]);
 
   useEffect(() => {
     const cancelTokenSource = axios.CancelToken.source();
+    setLoaging(true);
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:16]:
```suggestion
    setLoading(true);
```

File: client/src/App/index.jsx [hunk 3]
```diff
@@ -8,20 +8,23 @@ import '../style/custom-antd.css';
...
     axios
       .get('/api/v1/categories', {
         cancelToken: cancelTokenSource.token,
       })
       .then(({ data }) => {
+        setLoaging(false);
         setCategories(data.data);
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:23]:
```suggestion
      .then(({ data: { data } }) => {
        setCategories(data);
        setLoaging(false);
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/src/App/index.jsx [hunk 4]
```diff
@@ -8,20 +8,23 @@ import '../style/custom-antd.css';
...
       });
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:24]:
```suggestion
      })
      .catch(() => {
        setLoaging(false);
        message.error('حدث خطأ ما');
      });
```

File: client/src/Components/Empty/index.jsx [hunk 5]
```diff
@@ -1,7 +1,15 @@
...
 import { Empty } from 'antd';
 
-function EmptyCard() {
-  return <Empty description={false} />;
+function EmptyCard({description}) {
+  return <Empty description={
+    <span>
+      {description}
+    </span>
+  } />;
```
Comment (Mu7ammadAbed) [inline client/src/Components/Empty/index.jsx:10]: Why did you use the `span` ?
```suggestion
  return <Empty description={description} />;
```

File: client/src/Components/CategoriesCardsContainer/index.jsx [hunk 6]
```diff
@@ -28,14 +28,14 @@ function CategoriesCardsContainer({ categories, loading }) {
             ))}
           </Row>
         ) : (
-          <Empty />
+          <Empty description={"لا يوجد بيانات"} />
```
Comment (Mu7ammadAbed) [inline client/src/Components/CategoriesCardsContainer/index.jsx:31]:
```suggestion
          <Empty description="لا يوجد بيانات" />
```

File: client/src/App/index.jsx [hunk 7]
```diff
@@ -1,27 +1,30 @@
...
       .get('/api/v1/categories', {
         cancelToken: cancelTokenSource.token,
       })
-      .then(({ data }) => {
-        setLoaging(false);
-        setCategories(data.data);
-      });
+      .then(({ data: {data} }) => {
+        setCategories(data);
+        // setLoading(false);
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:23]:
```suggestion
        setLoading(false);
```

File: client/src/Components/CategoriesCardsContainer/index.jsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        ) : (
+          <Empty description={"لا يوجد بيانات"} />
+        )}
+      </div>
+    </>
+  );
+}
+CategoriesCardsContainer.propTypes = {
+  loading: PropTypes.bool,
+  categories: PropTypes.arrayOf(PropTypes.oneOfType(PropTypes.object))
```
Comment (Mu7ammadAbed) [inline client/src/Components/CategoriesCardsContainer/index.jsx:39]:
```suggestion
  categories: PropTypes.arrayOf(PropTypes.shape({}))
```

File: client/src/Components/CategoriesCardsContainer/index.jsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+          </Row>
+        ) : (
+          <Empty description={"لا يوجد بيانات"} />
+        )}
+      </div>
+    </>
+  );
+}
+CategoriesCardsContainer.propTypes = {
+  loading: PropTypes.bool,
```
Comment (Mu7ammadAbed) [inline client/src/Components/CategoriesCardsContainer/index.jsx:38]:
```suggestion
CategoriesCardsContainer.defaultProps = {
  loading: false,
};

CategoriesCardsContainer.propTypes = {
  loading: PropTypes.bool,
```

File: client/src/Components/Empty/index.jsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+}
+EmptyCard.propTypes = {
+  description: PropTypes.string.isRequired
+}
```
Comment (Mu7ammadAbed) [inline client/src/Components/Empty/index.jsx:14]:
```suggestion
EmptyCard.propTypes = {
  description: PropTypes.string
}

EmptyCard.defaultProps = {
  description: 'لا يوجد بيانات',
};
```

File: client/src/Pages/Home/index.jsx [hunk 11]
```diff
@@ -1,17 +1,21 @@
...
+      <CategoriesCardsContainer categories={categories} loading={loading} />
     </div>
   );
 }
+Home.defaultProps = {
+  loading: false,
+};
+Home.propTypes = {
+  loading: PropTypes.bool,
+  categories: PropTypes.arrayOf(PropTypes.oneOfType(PropTypes.object))
```
Comment (Mu7ammadAbed) [inline client/src/Pages/Home/index.jsx:18]:
```suggestion
  categories: PropTypes.arrayOf(PropTypes.shape({}))
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by MohammadAlHabil — APPROVED** [review]

**Review by MostafaQanoo — APPROVED** [review]

**Review by Jehad91 — APPROVED** [review]

**Pull request merged**